common/dbconfig.cpp

```diff
@@ -71,22 +75,22 @@ DBConfigPtr DBConfigManager::parseConfig(const Json::Value& root) {
       }
 
       if (isSet) {
-        newDBConfig.dataSetConfigMap[dataset] = config;
+        newDBConfig->dataSetConfigMap[dataset] = config;
       }
     }
   }
-  if (newDBConfig.dataSetConfigMap.size() == 0) {
+  if (newDBConfig->dataSetConfigMap.size() == 0) {
     // no data was loaded..
     return nullptr;
   }
-  return std::make_shared<const DBConfig>(std::move(newDBConfig));
+  return newDBConfig;
 }
 
 /* conf getter functions */
 ConfigPtr DBConfigManager::getConfig(const std::string& dbName) const {
+  folly::SharedMutex::ReadHolder holder(lock);
```

Comment:
this `ReadHolder` will only protect till we find the DBConfig for the `dbName`. Then, during the 102: `getReplicationMode` at `return config->replication_mode;`, it could still case the race condition. Should we consider return a copy of the DBConfig?

Reply:
ConfigPtr is a shared_ptr. Once it is returned, it will be valid , either it is from the previous or current config. It will be a valid data.